guidance/models/_azure_openai.py

```diff
@@ -74,6 +74,7 @@ def __init__(
                 model=model,
                 azure_endpoint=azure_endpoint,
                 api_key=api_key,
+                azure_ad_token_provider=azure_ad_token_provider,
```

Comment:
This didn't show up in the tests, since those rely on the `api_key` which was already being passed back and forth